src/module/EMPairProduction.cpp

```diff
@@ -190,11 +197,20 @@ void EMPairProduction::performInteraction(Candidate *candidate) const {
 	static PPSecondariesEnergyDistribution interpolation;
 	double Ee = interpolation.sample(E, s);
 	double Ep = E - Ee;
+	double f = Ep / E;
 
 	// sample random position along current step
 	Vector3d pos = random.randomInterpolatedPosition(candidate->previous.getPosition(), candidate->current.getPosition());
-	candidate->addSecondary(-11, Ee / (1 + z), pos);
-	candidate->addSecondary(11, Ep / (1 + z), pos);
+	double w0 = candidate->getWeight();
+	// apply sampling
+	if (random.rand() < pow(f, thinning)) {
```

Comment:
Here the probabilities are now the opposite for electrons and positrons compared to EMDoublePairProduction, right?

Is this by chance or is there a reason for it; making up for some of the charge imbalence?

Reply:
The convention doesn't really matter. The total energy is fixed. If the electron takes a fraction f of the primary energy (E), then it's energy is Ee, such that Ee+Ep=E.
If we choose a random number r and accept electrons if r<f^thinning, positrons are accepted if r<(1-f)^thinning. The only important requirement is to keep this consistency that for one particle we have f, for the other 1-f.

Reply:
Electrons or positrons are accepted based on random numbers, not on charge. The particle with the highest energy fraction tends to be added to the list. For thinning=1, if the electron takes 90% of the energy, it is likely that it will be added to the stack. In this case, the positron takes 10% of the energy, so it's less likely that it will be accepted.
If, on the other hand, the positron takes 90% of the energy and the electron 10%, it is the other way around.